src/RA_Defs.cpp

```diff
@@ -10,68 +10,150 @@ GetParseErrorFunc GetJSONParseErrorStr = GetParseError_En;
+    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>, wchar_t> converter;
     return converter.from_bytes(str);
 }
 
+_Use_decl_annotations_
 std::wstring Widen(const std::string& str)
 {
-    static std::wstring_convert< std::codecvt_utf8_utf16< wchar_t >, wchar_t > converter;
+    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>, wchar_t> converter;
     return converter.from_bytes(str);
 }
 
+_Use_decl_annotations_
 std::wstring Widen(const wchar_t* wstr)
 {
-    return std::wstring(wstr);
+    return std::wstring{ wstr };
 }
 
+_Use_decl_annotations_
 std::wstring Widen(const std::wstring& wstr)
 {
     return wstr;
 }
 
+_Use_decl_annotations_
 std::string Narrow(const char* str)
 {
-    return std::string(str);
+    return std::string{ str };
 }
 
+_Use_decl_annotations_
 std::string Narrow(const std::string& str)
 {
     return str;
 }
 
+_Use_decl_annotations_
+NativeString NativeStr(NonNativeCString str) noexcept
+{
+#if _MBCS
+    return Narrow(str);
+#elif _UNICODE 
+    return Widen(str);
+#else
+#error Invalid character set, only MBCS and Unicode are supported.
+#endif // _MBCS
+}
+
+_Use_decl_annotations_
+NativeString NativeStr(const NonNativeString& str) noexcept
+{
+#if _MBCS
+    return Narrow(str);
+#elif _UNICODE 
+    return Widen(str);
+#else
+#error Invalid character set, only MBCS and Unicode are supported.
+#endif // _MBCS
+}
+
+_Use_decl_annotations_
+NativeString NativeStr(NonNativeString&& str) noexcept
+{
+#if _MBCS
+    return Narrow(std::move(str));
+#elif _UNICODE 
+    return Widen(std::move(str));
+#else
+#error Invalid character set, only MBCS and Unicode are supported.
+#endif // _MBCS
+}
+
+_Use_decl_annotations_
+NativeString NativeStr(NativeString&& str) noexcept
+{
+#if _MBCS
+    return Narrow(std::move(str));
+#elif _UNICODE 
+    return Widen(std::move(str));
+#else
+#error Invalid character set, only MBCS and Unicode are supported.
+#endif // _MBCS
+}
+
+_Use_decl_annotations_
+NativeString NativeStr(const NativeString& str) noexcept
+{
+#if _MBCS
+    return Narrow(str);
+#elif _UNICODE 
+    return Widen(str);
+#else
+#error Invalid character set, only MBCS and Unicode are supported.
+#endif // _MBCS
+}
+
+_Use_decl_annotations_
+NativeString NativeStr(NativeCString str) noexcept
+{
+#if _MBCS
+    return Narrow(str);
+#elif _UNICODE 
+    return Widen(str);
+#else
+#error Invalid character set, only MBCS and Unicode are supported.
+#endif // _MBCS
+}
+
+_Use_decl_annotations_
 void RADebugLogNoFormat(const char* data)
 {
     OutputDebugString(NativeStr(data).c_str());
 
     //SetCurrentDirectory( g_sHomeDir.c_str() );//?
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, RA_LOG_FILENAME, "a") == 0)
-    {
-        fwrite(data, sizeof(char), strlen(data), pf);
-        fclose(pf);
-    }
+    // SAL didn't like this
+    std::unique_ptr<FILE, decltype(&std::fclose)> fp{ std::fopen(RA_LOG_FILENAME, "a"), std::fclose };
+    fwrite(data, sizeof(char), strlen(data), fp.get());
```

Comment:
`FILE* ` is never allowed to be `nullptr` according to the specification in stdio.h. I have a class wrapper for this already but probably will hold off on it until you guys accept #57